crates/nu-cli/src/completions/completion_common.rs

```diff
@@ -76,6 +76,13 @@ fn complete_rec(
                         completions.push(built);
                     }
                 }
+                // isdir tells us whether the current path has a trailing slash.
+                // Parsing a path string into a pathbuf loses that bit of information.
+                // We use isdir to confirm that the current component has at least one
+                // next component or a slash.
```

Comment:
I think this might be better suited as part of the doc comment for the function itself (and if you add that, could you add a note about what `dir` is too?)

Reply:
Apologies for being late with this one. I should have been more descriptive with `dir` so I renamed it. I also added doc comments as you stated.